### ISSUE TITLE
map_clone false positive unless `.cloned()` exists

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#443]
Currently the code just silently assumes that whatever has `.map(_)` also has `.cloned()`. However, checking for this without introducing false negatives is surprisingly tricky.

**Comment by malbarbo** [comment]
An example of false positive:

```rust
vec![1].iter_mut().map(|x| *x);
```

**Comment by mcarton** [comment]
I've looked at that one once, the `cloned` bound is incredibly tricky to check for.
@llogiq this is assigned to you, are you still planning to work on that?

**Comment by vks** [comment]
The wiki should probably be updated to mention the false positive under "know problems".

**Comment by oli-obk** [comment]
I'll give it another go, if that fails, we'll mark it in the docs
